interfaces.go

```diff
@@ -21,9 +24,16 @@ type collectorResponse interface {
 	Disable() bool
 }
 
+type reportRequest struct {
+	thriftRequest *lightstep_thrift.ReportRequest
+	protoRequest  *cpb.ReportRequest
+	httpRequest   *http.Request
+}
+
 // collectorClient encapsulates internal thrift/grpc transports.
 type collectorClient interface {
-	Report(context.Context, *reportBuffer) (collectorResponse, error)
+	Report(context.Context, reportRequest) (collectorResponse, error)
+	Translate(context.Context, *reportBuffer) (reportRequest, error)
```

Comment:
I think in the rebase my comments disappeared. I feel like it would be nice to split this into `collectorClient` and `collectorEncoder` (then we can unify the code across `grpcCollectorClient` and `httpCollectorClient` into `protoCollectorEncoder` and split `thriftCollectorClient` into the client and a new `thriftCollectorEncoder` or something like that.

Although the code definitely is pretty readable as is, so I don't think it's a big deal to leave it this way.

Mostly I am just thinking this is good because I know while brainstorming it was suggested that down the line we might want to have a more clear separation between an unencoded queue and an encoded queue and having these two things split would help that.

Reply:
Chatted offline. LGTM!